packages/astro/src/integration/index.ts

```diff
@@ -68,6 +69,20 @@ export const sentryAstro = (options: SentryOptions = {}): AstroIntegration => {
           options.debug && console.log('[sentry-astro] Using default server init.');
           injectScript('page-ssr', buildServerSnippet(options || {}));
         }
+
+        const isSSR = config && (config.output === 'server' || config.output === 'hybrid');
+        const shouldAddMiddleware = options.autoInstrumentation?.requestHandler !== false;
+
+        // Guarding calling the addMiddleware function because it was only introduced in astro@3.5.0
+        // Users on older versions of astro will need to add the middleware manually.
+        const supportsAddMiddleware = typeof addMiddleware === 'function';
+
+        if (supportsAddMiddleware && isSSR && shouldAddMiddleware) {
```

Comment:
Just thinking, will this break if people update the SDK & astro to the latest version and already have the middleware added manually - e.g. will it then add the middleware twice? or do middlewares (or our middleware) handle this scenario?

Reply:
Good point! 

> will this break if people update the SDK

Technically, we're still in alpha so I'd be fine with this. Furthermore, middleware wasn't yet documented anywhere except for the readme. 

However, for general robustness, we can probably detect and handle double wrapping by writing a flag to the `locals` object that's passed into the middleware handlers. I'll check take a look.

Reply:
if this is not documented yet, IMHO let's just break it! no need to introduce additional complexity then 👍